api/src/wfl/module/covid.clj

```diff
@@ -25,8 +18,65 @@
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
-(defn start-covid-workload!
-  [])
+;; TODO: move this to the queue-based skeleton later
+(defn ^:private snapshot-created?
+  "Check if `snapshot-job-id` has done in TDR,
+   return the resulting snapshot id if so."
+  [snapshot-job-id]
+  (some->> snapshot-job-id
+           (datarepo/job-done?)
+           :id))
+
+(defn ^:private go-create-snapshot!
+  "Create snapshot in TDR from `dataset`, `table`
+   and `row-ids` and write job info as well as rows
+   into `source-details-name` table."
+  [tx dataset table source-details-name [start end] row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (update :name util/randomize)
+                   ;; FIXME: this sometimes runs into: Cannot update iam permissions, need failure handling here!
+                   (datarepo/create-snapshot))]
+    (jdbc/insert! tx
+                  source-details-name
+                  {:snapshot_creation_job_id job-id
+                   :datarepo_row_ids row-ids
+                   :start_time start
+                   :end_time end})))
+
+(defn ^:private check-for-new-data!
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [tx
+   {:keys [source_item] :as workload}
+   {:keys [dataset table last_checked snapshot details] :as _source}]
+  (let [now (util/utc-now)
+        dataset (datarepo/dataset dataset)
+        table table
+        col-name snapshot
+        row-ids (-> (datarepo/query-table-between
+                     dataset
+                     table
+                     col-name
+                     [last_checked now]
+                     [:datarepo_row_id])
+                    :rows
+                    flatten)]
+    (when row-ids
+      (let [shards (partition-all 500 row-ids)]
+        (map #(go-create-snapshot! tx dataset table details [last_checked now] %) shards)))
+    (do
+      (jdbc/update! tx
+                    source_item
+                    {:last_checked now}
+                    ["id = ?" (:id workload)])
+      (jdbc/update! tx
+                    :workload
+                    {:updated (OffsetDateTime/now)}
+                    ["id = ?" (:id workload)]))))
```

Comment:
`"Create and record snapshot jobs in WORKFLOW with new data from SOURCE."`?
Why rename `snapshot` to `col-name` (and `table` to … `table`)?
Don't need `do`, I think.
What does `utc-now` add?
Maybe `java.time.Instant/now` instead, if necessary.
Can `row-ids` be `nil` or is it always a seq?
Let's talk about this in the afternoon session.